pkgs/tools/misc/execline/default.nix

```diff
@@ -43,29 +43,28 @@ let
 
     };
 
-  # a wrapper around execlineb, which provides all execline
+  # A wrapper around execlineb, which provides all execline
   # tools on `execlineb`’s PATH.
-  execlineb-with-builtins-drv =
-    let eldir = "${execline}/bin";
-    in writeScriptBin "execlineb" ''
-      #!${eldir}/execlineb -s0
-      # appends the execlineb bin dir to PATH if not yet in PATH
-      ${eldir}/define eldir ${eldir}
-      ''${eldir}/ifelse
-      {
-        # since this is nix, we can grep for the execline drv hash in PATH
-        # to see whether it’s already in there
-        ''${eldir}/pipeline
-        { ${coreutils}/bin/printenv PATH }
-        ${gnugrep}/bin/grep --quiet "${eldir}"
-      }
-      # it’s there already
-      { ''${eldir}/execlineb $@ }
-      # not there yet, add it
-      ''${eldir}/importas oldpath PATH
-      ''${eldir}/export PATH "''${eldir}:''${oldpath}"
-      ''${eldir}/execlineb $@
-    '';
+  # It is implemented as a C script, because on non-Linux,
+  # nested shebang lines are not supported.
+  execlineb-with-builtins-drv = runCommandCC "execlineb" {} ''
+    substitute \
```

Comment:
Now that it is a C programs, could we perhaps use constants and  `-D` flags?

Reply:
Good catch!